tests/clone_cell.rs

```diff
@@ -122,3 +151,66 @@ async fn clone_cell_management() {
         .await;
     assert!(enable_clone_cell_response.is_err());
 }
+
+// Check that app info can be refreshed to allow zome calls to a clone cell identified by its clone cell id
+#[tokio::test(flavor = "multi_thread")]
+pub async fn app_info_refresh() {
+    let conductor = SweetConductor::from_standard_config().await;
+    let admin_port = conductor.get_arbitrary_admin_websocket_port().unwrap();
+    let mut admin_ws = AdminWebsocket::connect(format!("ws://localhost:{}", admin_port))
+        .await
+        .unwrap();
+    let app_id: InstalledAppId = "test-app".into();
+    let role_name: RoleName = "foo".into();
+
+    // Create our agent key
+    let agent_key = admin_ws.generate_agent_pub_key().await.unwrap();
+
+    // Install and enable an app
+    admin_ws
+        .install_app(InstallAppPayload {
+            agent_key: agent_key.clone(),
+            installed_app_id: Some(app_id.clone()),
+            membrane_proofs: HashMap::new(),
+            network_seed: None,
+            source: AppBundleSource::Path(PathBuf::from("./fixture/test.happ")),
+        })
+        .await
+        .unwrap();
+    admin_ws.enable_app(app_id.clone()).await.unwrap();
+
+    let mut signer = ClientAgentSigner::default();
+    
+    // Create an app interface and connect an app agent to it
+    let app_api_port = admin_ws.attach_app_interface(30000).await.unwrap();
+    let mut app_agent_ws = AppAgentWebsocket::connect(format!("ws://localhost:{}", app_api_port), app_id.clone(), signer.clone().into()).await.unwrap();
+
+    // Create a clone cell, AFTER the app agent has been created
+    let cloned_cell = app_agent_ws.create_clone_cell(CreateCloneCellPayload {
+        app_id: app_id.clone(),
+        role_name: role_name.clone(),
+        modifiers: DnaModifiersOpt::none().with_network_seed("test seed".into()),
+        membrane_proof: None,
+        name: None,
+    }).await.unwrap();
+
+    // Authorise signing credentials for the cloned cell
+    let credentials = admin_ws.authorize_signing_credentials(AuthorizeSigningCredentialsPayload {
+        cell_id: cloned_cell.cell_id.clone(),
+        functions: None,
+    }).await.unwrap();
+    signer.add_credentials(cloned_cell.cell_id.clone(), credentials);
+
+    // Call the zome function on the clone cell, expecting a failure
+    let err = app_agent_ws.call_zome(cloned_cell.clone_id.clone().into(), "foo".into(), "foo".into(), ExternIO::encode(()).unwrap()).await.expect_err("Should fail because the client doesn't know the clone cell exists");
```

Comment:
this looks unformatted

Reply:
Good eye, missed formatting on the last push. Done now!